.github/workflows/yocto-build-deploy.yml

```diff
@@ -1192,3 +1192,40 @@ jobs:
           # Two variables extra for this job
           QEMU_SECUREBOOT: 1
           FLASHER_SECUREBOOT: 1
+
+  # This job always runs and will fail if any of the builds or tests fail.
+  # This way we can mark this job as required for merging PRs.
+  # Otherwise we would need to mark each build and test matrix, suite, etc. as required.
+  all_jobs:
+    name: All jobs
+    needs:
+      - build
+      - test
+    runs-on: ubuntu-latest
+    # Do not skip this job if needed jobs are skipped, as we always
+    # want to capture pass/fail for the entire workflow and skipped
+    # jobs count as passing as far as required checks go.
+    if: |
+      always() && inputs.run-tests == true
```

Comment:
Isn't `always()` redundant here?

Reply:
We need `always()` because the hidden defaults are `success()`, and we want to make sure this job always runs no matter the results of previous jobs.

If this job gets skipped because a previous job failed, GH would actually consider `All tests` to be passed and will merge the PR. I think they have their reasons for doing it that way but I don't recall what they are.

Reply:
I do have to remove `inputs.run-tests == true` though, that was before I included the build in this check.

Reply:
I've updated the comment!